internal/gh/rest.go

```diff
@@ -0,0 +1,219 @@
+
+    queryParameters := map[string]string{
+        "per_page" : "100",
+        "activity_type" : "force_push",
+        "ref" : branch,
+    }
+
+    headerParameters := map[string]string{
+        "Content-Type" : "application/json",
+        "Authorization": "Bearer "+token,
+    }
+
+    client := &http.Client{}
+
+    NumberForcePush, err := processForcePushRequest(client, repoActivityUrl, queryParameters, headerParameters) 
+    if err != nil {
+        slog.Default().Error("Getting repo info falied - %v getForcePushInfo method", err)
+        return 0, err
+    }    
+    return NumberForcePush, nil
+
+}
+
+//  Execute the HTTP request and return response
+func executeHTTPRequest(client *http.Client, req *http.Request) (*http.Response, error){
+
+    resp, err := client.Do(req)
+	RequestCounter++
+	if err != nil {
+        return nil, err
+	}
+    return resp, nil
+
+}
+
+//  Handle calls to other methods (createHTTPRequest, executeHTTPRequest, processHttpResponse) is handled and loop is added so that requests are processed till there no rel = "next"
+func processForcePushRequest(client *http.Client, repoActivityUrl string, queryParameters, headerParameters map[string] string) (int, error) {
+
+    NumberForcePush := 0
+    hasNext := true
+
+    // Create an initial HTTP Request and set header and query parameters
+    httpReq, err := createHttpRequest(client, repoActivityUrl,"GET", nil, queryParameters, headerParameters)
+    if err != nil {
+        slog.Default().Error("Failed to create HTTP request: %v", err)
+        return  0, err
+    }
+
+    for hasNext{
+
+        // Execute the HTTP Request
+        resp, err := executeHTTPRequest(client, httpReq)
+        if err != nil {
+		    slog.Default().Error("Failed to execute HTML request: %v - processForcePushRequest method", err)
+		    return 0, err
+	    }
+
+        defer resp.Body.Close()
+        
+        // Process the response as result data type
+        res, err := processHttpResponse(resp)
+        if err != nil {
+	        slog.Default().Error("Failed to process HTTP response: %v - processForcePushRequest method", err)
+	        return 0, err
+	    }
+
+        NumberForcePush += len(res)
+
+        // Check if Next Page exists         
+        hasNext, err = checkIfNextPageExists(resp)
+        if err != nil {
+	        slog.Default().Error("Failed to check for next pages in force pushes: %v - processHttpResponse method", err)
+	        return 0, err
+	    }        
+        // If next page exists get the url of the next page
+        if hasNext {
+
+            repoActivityUrl, err = getNextURL(resp)
+            if err != nil {
+	            slog.Default().Error("Failed to get Net Url: %v - processHttpResponse method", err)
+	            return 0, err
+	        }
+
+            // Update the exisiting http request
+            err := updateHttpRequest(repoActivityUrl, "GET", nil, queryParameters, headerParameters, httpReq)
```

Comment:
I believe you might have overdone this here a bit. I believe something like this might be sufficient, but please double check that:

```suggestion
            httpReq.URL = repoActivityUrl 
```

Reply:
Hey Jan, yeah I agree it is overdone but I wanted it to look clean and did not want to add statements in the same method.  
The problem here is that Http.Request.URL is not a string it is of the type url.URL which in it self is made up of the scheme, host path, etc. So we have to first parse the new url string and then endcode it and set it in the req Raw query. So I decided to go with a new method instead of adding those lines in the same method.

Reply:
oh, my bad I confused the APIs, I just checked.